api/src/database/migrations/20250514A-add-missing-retention-indexes.ts

```diff
@@ -0,0 +1,45 @@
+import type { Knex } from 'knex';
+import { getHelpers } from '../helpers/index.js';
+import { transaction } from '../../utils/transaction.js';
+import { FieldsService } from '../../services/fields.js';
+import { getSchema } from '../../utils/get-schema.js';
+import { getDatabaseClient } from '../index.js';
+
+const RETENTION_INDEXES = [
+    { collection: 'directus_activity', field: 'timestamp', ignore: [] },
+    { collection: 'directus_revisions', field: 'parent', ignore: ['mysql'] },
+];
+
+export async function up(knex: Knex): Promise<void> {
+    const client = getDatabaseClient(knex);
+    const helpers = getHelpers(knex);
+    const schema = await getSchema();
+    const service = new FieldsService({ knex, schema });
+
+    for (const { collection, field, ignore } of RETENTION_INDEXES) {
+        const existingColumn = await service.columnInfo(collection, field);
+
+        if (!existingColumn.is_indexed && !ignore.includes(client)) {
```

Comment:
If we move the client ignore check before the `columnInfo` call we can reduce the db queries for ignored clients